koku/subs/tasks.py

```diff
@@ -0,0 +1,117 @@
+#
+# Copyright 2023 Red Hat Inc.
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Tasks for Subscriptions Watch"""
+import datetime
+import logging
+import uuid
+
+from dateutil import parser
+
+from api.common import log_json
+from api.provider.models import Provider
+from hcs.tasks import get_start_and_end_from_manifest_id
+from koku import celery_app
+from koku import settings
+from koku.feature_flags import fallback_development_true
+from koku.feature_flags import UNLEASH_CLIENT
+from masu.external.date_accessor import DateAccessor
+from masu.util.common import convert_account
+from subs.subs_data_extractor import SUBSDataExtractor
+from subs.subs_data_messenger import SUBSDataMessenger
+
+LOG = logging.getLogger(__name__)
+
+SUBS_EXTRACTION_QUEUE = "subs_extraction"
+SUBS_TRANSMISSION_QUEUE = "subs_transmission"
+
+# any additional queues should be added to this list
+QUEUE_LIST = [SUBS_EXTRACTION_QUEUE, SUBS_TRANSMISSION_QUEUE]
+
+SUBS_ACCEPTED_PROVIDERS = (
+    Provider.PROVIDER_AWS,
+    Provider.PROVIDER_AWS_LOCAL,
+    # Add additional accepted providers here
+)
+
+
+def check_subs_source_gate(schema_name: str) -> bool:
+    """Checks if the specific source is selected for RHEL Metered processing."""
+    # TODO: COST-4033 implement sources gate
+    return False
+
+
+def enable_subs_extraction(schema_name: str) -> bool:
+    """Helper to determine if source is enabled for SUBS extraction."""
+    schema_name = convert_account(schema_name)
+    context = {"schema_name": schema_name}
+    LOG.info(log_json(msg="enable_subs_extraction context", context=context))
+    return bool(
+        UNLEASH_CLIENT.is_enabled("cost-management.backend.subs-data-extraction", context, fallback_development_true)
+        or settings.ENABLE_SUBS_DEBUG
+        or check_subs_source_gate(schema_name)
+    )
+
+
+def enable_subs_messaging(schema_name: str) -> bool:
+    """Helper to determine if source is enabled for SUBS messaging."""
+    schema_name = convert_account(schema_name)
+    context = {"schema_name": schema_name}
+    LOG.info(log_json(msg="enable_subs_messaging context", context=context))
+    return bool(
+        UNLEASH_CLIENT.is_enabled("cost-management.backend.subs-data-messaging", context) or settings.ENABLE_SUBS_DEBUG
+    )
+
+
+@celery_app.task(name="subs.tasks.collect_subs_report_data_from_manifest", queue=SUBS_EXTRACTION_QUEUE)
+def collect_subs_report_data_from_manifest(reports_to_subs_summarize):
```

Comment:
Given that this task goes to the `EXTRACTION` queue, is this task "extracting" data?

```suggestion
def extract_subs_data_from_reports(reports_to_extract):
```

Reply:
Updated name appropriately.